### PR TITLE
Document effects of libtorch asynchronous execution on backend

### PR DESCRIPTION
**Pull request by tanmayv25**
Documenting current behavior for now. Should be updated with tritonserver enhancement to detect cuda context corruption.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,3 +120,14 @@ key: "DISABLE_OPTIMIZED_EXECUTION"
     }
 }
 ```
+
+### Important Note
+
+The execution of pytorch model on GPU is asynchronous in nature. See
+[here](https://pytorch.org/docs/stable/notes/cuda.html#asynchronous-execution) for more details.
+Consequently, an error in pytorch model execution may be raised in the next few inference requests to the
```
Comment (CoderHam) [inline README.md:128]: raised _during_ the next ...?

File: README.md [hunk 2]
```diff
@@ -120,3 +120,14 @@ key: "DISABLE_OPTIMIZED_EXECUTION"
...
+server. Setting environment variable `CUDA_LAUNCH_BLOCKING=1` when launching server will help in correctly
+debugging failing cases by forcing synchronous execution.
+
+The PyTorch model in such cases may or may not recover from the failed state and a restart of application
```
Comment (CoderHam) [inline README.md:132]: restart of **the server**

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Comment by CoderHam** [comment]
@deadeyegoodwin we should also call our the multiple instance perf issues with the pytorch backend due to thread specific caching in pytorch. Do you think it would be a good idea to have this here?

**Comment by deadeyegoodwin** [comment]
> 
> 
> @deadeyegoodwin we should also call our the multiple instance perf issues with the pytorch backend due to thread specific caching in pytorch. Do you think it would be a good idea to have this here?

Yes, it would be good to refer to that open issue as well.

**Comment by CoderHam** [comment]
@tanmayv25 do you want to add in the content in or prefer I do that in a follow up PR?
PS: PyTorch issue https://github.com/pytorch/pytorch/issues/27902

**Comment by tanmayv25** [comment]
@CoderHam  I think you would be able to better document it with a follow-up PR.. I will bullet these points so we can point to various issues observed with the framework.

**Comment by CoderHam** [comment]
Sounds good :)

**Review by CoderHam — APPROVED** [review]

**Review by CoderHam — APPROVED** [review]

**Pull request merged**